warehouse/router/sync.go

```diff
@@ -0,0 +1,81 @@
+package router
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	obskit "github.com/rudderlabs/rudder-observability-kit/go/labels"
+	"github.com/rudderlabs/rudder-server/warehouse/integrations/manager"
+	"github.com/rudderlabs/rudder-server/warehouse/internal/model"
+	"github.com/rudderlabs/rudder-server/warehouse/schema"
+	warehouseutils "github.com/rudderlabs/rudder-server/warehouse/utils"
+)
+
+type syncSchemaRepo interface {
+	GetLocalSchema(ctx context.Context) (model.Schema, error)
+	UpdateLocalSchemaWithWarehouse(ctx context.Context, schema model.Schema) error
+	HasSchemaChanged(schema model.Schema) bool
+	FetchSchemaFromWarehouse(ctx context.Context, m schema.FetchSchemaRepo) (model.Schema, error)
+}
+
+func (r *Router) sync(ctx context.Context) error {
```

Comment:
@cisse21 Thinking of tracking below list as a backlog task:
1. Metric for failures
2. How long it takes for a schema sync job to run for all warehouse models
3. Consider if we would want any parallelism more so for MT or even hosted.

CC: @achettyiitr

Reply:
Will add metrics... didn't add parallelism in the first iteration as we plan to sync for every 12hrs or more